### PR TITLE
Change code snippets from python 2 to python 3 in the documentation

### LINKED ISSUE #3582: Change code snippets from python 2 to python 3 in the documentation
This issue can be open for multiple PRs.

in the 'Overview over MDAnalysis' [page](https://docs.mdanalysis.org/2.1.0-dev0/documentation_pages/overview.html) of the user guide, some leftover python 2 syntax should be updated to python 3 syntax.

Example: `>>> print u.atoms`  should be replaced by  `>>> print(u.atoms)` 

Find any other obsolete code snippets in other pages of the user guide to add to this issue.

### PR DESCRIPTION
**Pull request by HenokB**
Fixes #3582 

Changes made in this Pull Request:
 - Change code snippets from python 2 to python 3 in the documentation
 - Update print statements from python 2 to python 3 

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by orbeckst** [comment]
@HenokB the testsuite has to run first. I approved the first run (that's currently required for new contributors). Once all tests are run, the coverage should be ok.

Could you please edit your PR text to reference the issue number so that it gets linked to the issue? Thanks.

**Comment by HenokB** [comment]
> @HenokB the testsuite has to run first. I approved the first run (that's currently required for new contributors). Once all tests are run, the coverage should be ok.
> 
> Could you please edit your PR text to reference the issue number so that it gets linked to the issue? Thanks.

Done

**Comment by RMeli** [comment]
Probably outside the scope of this PR, but I think it would be nice to remove the `from MDAnalysis import *` as well (and change them to `from MDAnalysis import Universe` where needed; this is already done in one case, so the change would also uniform things).

**Comment by RMeli** [comment]
@HenokB looks like there are still a few `print` statements that you have not changed.

**Comment by HenokB** [comment]
> @HenokB looks like there are still a few `print` statements that you have not changed.

Yeah, I know that. I read somewhere that it's a good practice to do a few changes at a time and I don't want to mess up things. I will change the rest of the print statements in one PR.

**Comment by IAlibay** [comment]
> > @HenokB looks like there are still a few `print` statements that you have not changed.
>
> Yeah, I know that. I read somewhere that it's a good practice to do a few changes at a time and I don't want to mess up things. I will change the rest of the print statements in one PR.

Just to expand on this, when we mean "few changes at a time" we usually mean the difference between 200 lines vs 2000 lines. It's all about readability and the amount of work it takes us to review. In the case of docstring changes it's fine to change multiple print statements since it's something that's easy for us to look at and say "yeah that's how I expect print to work"

**Comment by HenokB** [comment]
> > > @HenokB looks like there are still a few `print` statements that you have not changed.
> > 
> > 
> > Yeah, I know that. I read somewhere that it's a good practice to do a few changes at a time and I don't want to mess up things. I will change the rest of the print statements in one PR.
> 
> Just to expand on this, when we mean "few changes at a time" we usually mean the difference between 200 lines vs 2000 lines. It's all about readability and the amount of work it takes us to review. In the case of docstring changes it's fine to change multiple print statements since it's something that's easy for us to look at and say "yeah that's how I expect print to work"

Okay, got it. I will do a new PR with all the print statements fixed.

**Comment by RMeli** [comment]
> I read somewhere that it's a good practice to do a few changes at a time and I don't want to mess up things.

To add on @IAlibay comment, this good practice sometimes refers to commits. You should do small commits and commit often (to keep track of changes and not loose any work), but a PR can contain multiple commits. Yet another way of reading this good practice is that a PR should be focused (fix one and only one issue or, provide only one enhancement, ...).

> Okay, got it. I will do a new PR with all the print statements fixed.

You don't need to do a new PR (and you shouldn't). You can fix the remaining `print` statements on your `patch-2` branch, `commit` them, and `push` them. They will appear in this PR (since a PR tracks the changes on the branch it was started from).

---

Edit: Refreshing the page I see you already did this, that's awesome!

**Comment by HenokB** [comment]
> 
I actually did that, thanks

**Comment by HenokB** [comment]
> > I read somewhere that it's a good practice to do a few changes at a time and I don't want to mess up things.
> 
> To add on @IAlibay comment, this good practice sometimes refers to commits. You should do small commits and commit often (to keep track of changes and not loose any work), but a PR can contain multiple commits. Yet another way of reading this good practice is that a PR should be focused (fix one and only one issue or, provide only one enhancement, ...).
> 
> > Okay, got it. I will do a new PR with all the print statements fixed.
> 
> You don't need to do a new PR (and you shouldn't). You can fix the remaining `print` statements on your `patch-2` branch, `commit` them, and `push` them. They will appear in this PR (since a PR tracks the changes on the branch it was started from).
> 
> Edit: Refreshing the page I see you already did this, that's awesome!

Is it ready to get merged now?

**Comment by HenokB** [comment]
This issue got all the checks correct, is it possible to merge it now?

**Comment by orbeckst** [comment]
Someone will review your changes and comment. If they find anything they will request changes. Otherwise they will approve and then merge. Please be patient, in can take a day until someone can make the time to review.

File: package/doc/sphinx/source/documentation_pages/overview.rst [hunk 1]
```diff
@@ -35,7 +35,7 @@ of a protein and the radius of gyration of the backbone atoms are calculated::
         r = cterm.pos - nterm.pos  # end-to-end vector from atom positions
         d = numpy.linalg.norm(r)   # end-to-end distance
         rgyr = bb.radius_of_gyration()  # method of a AtomGroup; updates with each frame
-        print "frame = %d: d = %f Angstroem, Rgyr = %f Angstroem" % (ts.frame, d, rgyr)
+        print("frame = %d: d = %f Angstroem, Rgyr = %f Angstroem" % (ts.frame, d, rgyr))
```
Comment (micaela-matta) [inline package/doc/sphinx/source/documentation_pages/overview.rst]: while you're here can you turn this print statement into an [f-string](https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python)?

**Review by micaela-matta — CHANGES_REQUESTED** [review]

**Review by orbeckst — CHANGES_REQUESTED** [review]
There are more instances of old python 2 in the docs. Some of them you can find with
```
cd package/doc/sphinx/source
git grep -e 'print\s* [^(]'
```
which gives
```
documentation_pages/overview.rst:        print "frame = %d: d = %f Angstroem, Rgyr = %f Angstroem" % (ts.frame, d, rgyr)
documentation_pages/overview.rst:  >>> print u.atoms
documentation_pages/overview.rst:  >>> print u.trajectory
documentation_pages/overview.rst:  >>> print calphas
documentation_pages/selections.rst: >>> print list(ordered)
documentation_pages/selections.rst: >>> print list(universe.select_atoms("segid DMPC and resid 3 and name P", "segid DMPC and resid 2 and name P"))
documentation_pages/selections.rst: >>> print list(universe.select_atoms("segid DMPC and ( resid 3 or resid 2 ) and name P"))
```

Look at the docs in these files. Some of them you already fixed (good!). If the `prints` are Python 2 then there's a good chance that other code is also outdated. We like it when applicants show initiative and investigate by themselves.

Please fix at a minimum the other instances of Python 2 code that I flagged above.

**Comment by HenokB** [comment]
> There are more instances of old python 2 in the docs. Some of them you can find with
> 
> ```
> cd package/doc/sphinx/source
> git grep -e 'print\s* [^(]'
> ```
> 
> which gives
> 
> ```
> documentation_pages/overview.rst:        print "frame = %d: d = %f Angstroem, Rgyr = %f Angstroem" % (ts.frame, d, rgyr)
> documentation_pages/overview.rst:  >>> print u.atoms
> documentation_pages/overview.rst:  >>> print u.trajectory
> documentation_pages/overview.rst:  >>> print calphas
> documentation_pages/selections.rst: >>> print list(ordered)
> documentation_pages/selections.rst: >>> print list(universe.select_atoms("segid DMPC and resid 3 and name P", "segid DMPC and resid 2 and name P"))
> documentation_pages/selections.rst: >>> print list(universe.select_atoms("segid DMPC and ( resid 3 or resid 2 ) and name P"))
> ```
> 
> Look at the docs in these files. Some of them you already fixed (good!). If the `prints` are Python 2 then there's a good chance that other code is also outdated. We like it when applicants show initiative and investigate by themselves.
> 
> Please fix at a minimum the other instances of Python 2 code that I flagged above.

Thanks, and yeah I fixed the other python-2 statements you flagged in on the > documentation_pages/selections.rst: page

**Comment by RMeli** [comment]
@HenokB , as we discussed in https://github.com/MDAnalysis/mdanalysis/pull/3588#issuecomment-1084326318, you should do all related changes in the same branch (`pathch-2` in this case), which will show up in this PR. There is no need to open a PR for every change.

You should move the changes you did in PR #3593 here.

**Comment by RMeli** [comment]
Here you can also follow Lily's suggestion (https://github.com/MDAnalysis/mdanalysis/pull/3587#issuecomment-1086612901) and do
```
git checkout patch-2
git merge patch-3
```
in order to bring the changes you did in https://github.com/MDAnalysis/mdanalysis/pull/3593 into this PR.

**Comment by HenokB** [comment]
> Here you can also follow Lily's suggestion ([#3587 (comment)](https://github.com/MDAnalysis/mdanalysis/pull/3587#issuecomment-1086612901)) and do
> 
> ```
> git checkout patch-2
> git merge patch-3
> ```
> 
> in order to bring the changes you did in #3593 into this PR.

I did, thanks.

**Comment by HenokB** [comment]
Btw, since there is a code change here, should I add the fixes on the CHANGELOG?

**Comment by orbeckst** [comment]
There are no executable code changes, only docs. Add your name to the CHANGELOG but don't add a detailed entry for a fix as it will not affect people executing code.  (And add your name to AUTHORS if your name isn't in there already).

**Comment by lilyminium** [comment]
> Add your name to the CHANGELOG but don't add a detailed entry for a fix as it will not affect people executing code. (And add your name to AUTHORS if your name isn't in there already).

I agree with @orbeckst here -- however, as you already have #3587 that adds your name to the CHANGELOG and is likely to get merged before this one, I think that will suffice :)

**Comment by HenokB** [comment]
> There are no executable code changes, only docs. Add your name to the CHANGELOG but don't add a detailed entry for a fix as it will not affect people executing code. (And add your name to AUTHORS if your name isn't in there already).

**Comment by HenokB** [comment]
> > Add your name to the CHANGELOG but don't add a detailed entry for a fix as it will not affect people executing code. (And add your name to AUTHORS if your name isn't in there already).
> 
> I agree with @orbeckst here -- however, as you already have #3587 that adds your name to the CHANGELOG and is likely to get merged before this one, I think that will suffice :)

Alright, in that case I think it's ready to be merged. Btw I accidentally closed the PR, but I reopened it again.

File: package/doc/sphinx/source/documentation_pages/selections.rst [hunk 2]
```diff
@@ -405,22 +405,22 @@ The most straightforward way to concatentate two AtomGroups is by using the
...
+ >>> print(list(universe.select_atoms("segid DMPC and resid 3 and name P", "segid DMPC and resid 2 and name P")))
  [< Atom 570: name 'P' of type '180' of resid 'DMPC', 3 and 'DMPC'>,
  < Atom 452: name 'P' of type '180' of resid 'DMPC', 2 and 'DMPC'>]
 
 Just for comparison to show that a single selection string does not
 work as one might expect::
 
  # WRONG!
- >>> print list(universe.select_atoms("segid DMPC and ( resid 3 or resid 2 ) and name P"))
+ >>> print(list(universe.select_atoms("segid DMPC and ( resid 3 or resid 2 ) and name P")))
```
Comment (orbeckst) [inline package/doc/sphinx/source/documentation_pages/selections.rst]: We can remove the spaces after `(` and before `)`
```suggestion
 >>> print(list(universe.select_atoms("segid DMPC and (resid 3 or resid 2) and name P")))
```
This was a left-over from a _very_ long time ago (>10 years??) when the selection parser could not deal with parentheses properly.

You can just "commit the suggestion" as is.

**Review by orbeckst — APPROVED** [review]
Looks good to me. I have only one non-blocking suggestion to clean up the examples further.

**Comment by orbeckst** [comment]
... and you'll have to resolve conflicts. Please ask if you need help.

**Comment by orbeckst** [comment]
@micaela-matta can you please finalize the PR? I think it's good to be squash-merged.

**Review by micaela-matta — APPROVED** [review]

**Pull request merged**